### PR TITLE
Float values serialization tests

### PR DESCRIPTION
**Pull request by vitvakatu**
- Added license to the new `float` module
- Fixed some doc tests in `float` module
- Added tests for the following cases: zero, subnormal (denormal) number, normal number, maximum and minimum numbers, +/- infinity, NaNs.

### CODE REVIEW

**Comment by stanislav-tkach** [comment]
Please review: @vldm, @alekseysidorov, @aleksuss, @DenisKolodin, @ozkriff. :bell:

File: exonum/src/encoding/tests.rs [hunk 1]
```diff
@@ -144,6 +148,48 @@ fn test_i64_segment() {
     assert_write_check_read(dat, 8);
 }
 
+#[cfg(feature="float_serialize")]
```
Comment (stanislav-tkach) [inline exonum/src/encoding/tests.rs]: Maybe this should be moved into the `float` module too?..

@vldm What do you think?

**Review by stanislav-tkach — APPROVED** [review]

**Comment by vitvakatu** [comment]
I've moved tests to the `float` module and rewritten them a bit.
Also I changed wrapper's constructors in order to forbid subnormal numbers.
Requesting review from @vldm

**Review by stanislav-tkach — APPROVED** [review]

**Comment by vitvakatu** [comment]
Do I need to rebase?

File: exonum/src/encoding/float.rs [hunk 2]
```diff
@@ -244,3 +258,74 @@ impl ExonumJson for F64 {
...
+            assert_eq!(validate_constructor(|v| F32::new(v), value, &buf, 4), ValidationResult::Valid);
+        }
+        for value in invalid_data {
+            LittleEndian::write_f32(&mut buf, value);
+            assert_eq!(validate_constructor(|v| F32::new(v), value, &buf, 4), ValidationResult::Invalid);
+        }
+    }
+
+    #[test]
+    fn test_f64_segment() {
```
Comment (vldm) [inline exonum/src/encoding/float.rs]: This test is for f32 value, not segment of f32

**Review by vldm — APPROVED** [review]

**Comment by vitvakatu** [comment]
Do not merge for now, needs rebase

File: exonum/src/encoding/float.rs [hunk 3]
```diff
@@ -65,10 +66,9 @@ impl F32 {
     /// assert!(val.is_none());
     /// ```
     pub fn try_from(value: f32) -> Option<Self> {
-        if value.is_normal() {
-            Some(Self { value })
-        } else {
-            None
+        match value.classify() {
+            FpCategory::Normal | FpCategory::Zero => Some(Self { value }),
+            _ => None,
```
Comment (stanislav-tkach) [inline exonum/src/encoding/float.rs:71]: I propose to move this check into a function.

File: exonum/src/encoding/float.rs [hunk 4]
```diff
@@ -270,7 +269,13 @@ mod tests {
     use byteorder::{LittleEndian, ByteOrder};
     use encoding::Offset;
 
-    fn assert_validate_constructor<T, V, C: Fn(V) -> T>(constructor: C, value: V, buffer: &[u8], header_size: Offset)
+    #[derive(Debug, PartialEq)]
+    enum ValidationResult {
```
Comment (stanislav-tkach) [inline exonum/src/encoding/float.rs]: Is it worth creating an enum for that?

File: exonum/src/encoding/float.rs (hunk 3 above)
Comment (vitvakatu) [reply exonum/src/encoding/float.rs:71]: I don't think it worth it.

File: exonum/src/encoding/float.rs (hunk 4 above)
Comment (vitvakatu) [reply exonum/src/encoding/float.rs]: Will replace with bool

**Pull request merged**